### ISSUE TITLE
Using feathers-authentication-client for an existing API?

### ISSUE THREAD

**Issue by prime-ralph** [feathersjs/feathers-authentication#478]
### Steps to reproduce
Was doing this in a fresh project: 
```
import feathers from 'feathers/client';
import superagent from 'superagent';
import hooks from 'feathers-hooks';
import client from 'feathers-rest/client';
import localstorage from 'localstorage-memory';
import auth from 'feathers-authentication-client';

const apiRest = client(process.env.API_ENDPOINT);
const api = feathers()
  .configure(hooks())
  .configure(auth({
    storage: localstorage,
    header: 'Authorization',
    path: process.env.AUTH_SERVICE,
    type: 'jwt',
    storageKey: 'token',
  }))
  .configure(apiRest.superagent(superagent));

api.authenticate({
  strategy: 'local',
  user_name: '...',
  password: '...',
})
.then((response) => {
  console.log('Authenticated!', response, api.get('token'));
  return api.passport.verifyJWT(response.token);
}).then((payload) => {
  console.log('JWT Payload', payload);
  return '';
});
```

### Actual behavior
Instead of the behavior outlined in the docs, this results in the following:
`api.get('token')` returns undefined, and the localstorage token has a key of 'token', but with a value of `[object Object]`.

I suspect this is because I'm trying to authenticate with a different API server (a Laravel jwt-auth one instead of another Feathers server) which returns, upon authentication, `{ token: '...'}`.

At the `console.log('Authenticated!', response, api.get('token'));` part, the response prints out while the `api.get()` call returns undefined. If I try another request in that promise chain, I can also see that the token isn't being included in subsequent call headers.

### System configuration

Tell us about the applicable parts of your setup.

**Module versions** (especially the part that's not working):
feathers@2.1.1
feathers-authentication-client@0.3.1
feathers-hooks@1.8.1
feathers-rest@1.7.1

**NodeJS version**:
v6.9.1

**Operating System**:
Windows 10

**Browser Version**:
Latest Chrome.

**Module Loader**:
Basic `react-boilerplate` stuff.
webpack@2.2.0-rc.3
webpack-dev-middleware@1.9.0
webpack-hot-middleware@2.15.0

**Comment by marshallswain** [comment]
What version of feathers-authentication is on the existing API?

**Comment by prime-ralph** [comment]
I'm sorry. We were using a Laravel API in a /authenticate route that 
returns the token in the format of `{ token: '...' }`. Since it was 
authenticating and the token was being returned, I thought the problem 
was due to feathers-authentication on the client not knowing the key to 
take. Is there a way to configure it as such?

**Comment by marshallswain** [comment]
It's not configurable right now.  Here's the line that needs to be modified: https://github.com/feathersjs/feathers-authentication-client/blob/master/src/passport.js#L235

**Comment by prime-ralph** [comment]
So the whole { token: '...' } is stored because there's no `accessToken` field? 
Thanks for the heads-up. Might consider adding a custom authentication route on our API for this.

**Comment by marshallswain** [comment]
We really ought to allow this to be customized.  I created an issue for it.  I don't think it would be a difficult PR if you want to give it a shot.  ;)

**Comment by prime-ralph** [comment]
Well, since you've brought it up, might as well try. :)

**Comment by ekryski** [comment]
Ok, going to close this in favour of tracking in the auth-client issue.

**Comment by ekryski** [comment]
@prime-ralph I also added a comment here: https://github.com/feathersjs/feathers-authentication-client/issues/38#issuecomment-293154464

TL;DR could you not use a hook on the client to transform to and from the key name that you want?

**Comment by prime-ralph** [comment]
A question: from what I gather, you can only use hooks on services, but since I'm using the `authenticate` function, is there a way to bind a hook on that?

**Comment by marshallswain** [comment]
```js
app.service('authentication').hooks(`...`)
```

**Comment by prime-ralph** [comment]
After tuning, here's what I ended up using, for those who end up with the same situation:
```
api.service('authenticate').hooks({
  after: {
    all(hook) {
      Object.assign(hook.result, { accessToken: `Bearer ${hook.result.token}` });
      return Promise.resolve(hook);
    },
  },
});
```

The `Bearer` prefix was added because apparently our API requires the format of `Authorization: Bearer ...`.
Not sure if that's the cleanest I could do, just following the linter.
